docs/serialization.rst

```diff
@@ -260,6 +257,13 @@ Note that, when an old reader reads data written by an old writer, or a new
 reader reads a data written by a new writer, they will be able to read all
 fields.
 
+One thing to be careful while evolving the class is to not have any conditional
```

Comment:
In the `get_type_name` 's documentation there's some info about using the same type name between evolved types, should we also mention that here?

Reply:
https://github.com/hazelcast/hazelcast-python-client/pull/611/commits/66d9811255e30a1abc955f72957793e2b9cbedb0